lib/Core/Handler/HandlerMethodRunner.php

```diff
@@ -98,7 +98,19 @@ public function dispatch(Message $request): Promise
                 return null;
             }
 
-            return new ResponseMessage($request->id, yield $promise);
+            try {
+                $return = yield $promise;
+            } catch (\Throwable $exception) {
+                $this->logger->error(sprintf(
+                    "Method '%s' failed with exception: %s",
+                    $method,
+                    $exception->__toString()
+                ));
```

Comment:
this should already be handled in the logging middleware (the error will bubble up and be caught there)

Reply:
https://github.com/phpactor/language-server/blob/413b6580bef77bbb46b824760105dd0aec7919d7/lib/Middleware/ErrorHandlingMiddleware.php#L56-L77

Reply:
Hmm, okay. Without this I have not seen any logs in log-file, so that's why the PR. I'll have a look later why this didn't work for me

Reply:
Okay I see 2 issues now:

1. On that mentioned ErrorHandlingMiddleware the log is written only if request is not RequestMessage. I guess message could be written in either case?

2. RequestMessage sent back to LSP client is not valid as per JSON RPC 2.0 - https://www.jsonrpc.org/specification#response_object  
Property `result` MUST NOT exist if there was an error invoking the method. Which is why my LSP client didn't show me any problems either

Reply:
for 1 :+1: 
for 2: what is `result` set to then? `null` fields should be excluded from the response already IIRC.

Reply:
for 2: Created pr #50 - It's beeing set to `null` and it's not excluded

Reply:
Updated this PR for 1.
Failing check is due to code not related to this PR